### DISCUSSION TITLE
Cannot create a macOS vm using apple script

### DISCUSSION THREAD

**Discussion by gnasr** [utmapp/UTM#5642] [Development]
**Describe the issue**  
I'm trying to script a macOS vm creation following the example in the CheatSheet but the resulting VM is always a linux VM, the documentation seems to be incomplete too, since it doesn't show how to specify boot or operating system parameters in configuration (if it's possible at all).
I don't know if i understand correctly but it seems to be hardcoded that the VM created with scripting will always be linux, am i right? https://github.com/utmapp/UTM/blob/main/Scripting/UTMScriptingCreateCommand.swift#L106

**Configuration**  
* UTM Version: 4.2.5
* macOS Version: 13.3.1
* Mac Chip:  M1

If someone can show me how to script a macOS vm would be really helpful.
Thanks

**Comment by gustaf71** [comment]
In AppleScript property of object could have property of other object. In other words a nested object.
In normal case the specified object that contents property you use the same method set property of object to value.
UTM have a special method to update configuration.

Here is example how to change memory for VM from 8GB to 4GB.
```
tell application "UTM"
	set theVMs to name of virtual machines
	if "Fedora 38 ARM" is in theVMs then
		tell virtual machine "Fedora 38 ARM"
			log {"Current memory: ", memory of (get configuration)}
			set newConf to {memory:4096}
			update configuration with newConf
		end tell
	end if
end tell
```

To make is more clear you need to understand how the nested configuration of specified object work.
```
set thePath to POSIX path of (path to downloads folder) & "kali-linux-2023.2a-installer-arm64.iso"
set theIso to POSIX file thePath
tell application "UTM"
	set theDrive to {{removable:true, source:theIso}, {guest size:65536}}
	set theConfiguration to {name:"QEMU ARM64", architecture:"aarch64", drives:theDrive}
	set theVM to make new virtual machine with properties {backend:qemu, configuration:theConfiguration}
end tell

(*
{
   backend:qemu, 
   configuration:
   {
      name:"QEMU ARM64", 
      architecture:"aarch64", 
      drives:
      {
         {
            removable:true, 
            source:iso
         }, 
         {
            guest size:65536
         }
      }
   }
}
 *)

```

The drives object has 2 AppleScript record object.
```
tell application "UTM"
	tell virtual machine "Fedora 38 ARM"
		set theConfiguration to its configuration
		tell theConfiguration
			set {recordObject1, recordObject2} to {item 1 of its drives, item 2 of its drives}
			-- Return AppleScript record of object 1
			return recordObject1
			-- Return AppleScript record of object 2
			return recordObject2
		end tell
	end tell
end tell
```

**Comment by opensource-usr** [comment]
Hello, any update/comment on this issue please ? I am facing the same issue and not able to change "operating system" under "boot" options from "Linux" to "macOC" using the **Apple Script**.

I do not see any option under ["apple configuration" section in the documentation](https://docs.getutm.app/scripting/reference/#UTM-Configuration-Suite.apple-configuration)  as well.

I tried below code to change the OS configuration but resulted into an error.

`
tell application "UTM"

	set iso to POSIX file "/path/UniversalMac_12.0.1_21A559_Restore.ipsw"

	set theDrive to {{removable:true, source:iso}, {guest size:65536}}

	set theBoot to {{operatingSystem:"macOs"}}

	set theConfiguration to {name:"macOS-Name", boot:theBoot, drives:theDrive}

	make new virtual machine with properties {backend:apple, configuration:theConfiguration}

end tell

`

Error: "error "UTM got an error: Can’t make "

Thanks for your help in advance !

**Comment by verytheoretical** [comment]
It appears that currently the `apple` backend is hardcoded to create a VM of type `linux`, there is currently no functionality for creating a MacOS VM via the scripting API.

**Comment by naveenrajm7** [comment]
I have created an issue for this #6982 , I was able do fix this locally and have suggested the same in the report.
